torch/csrc/jit/codegen/cuda/kernel_cache.cpp

```diff
@@ -204,35 +204,45 @@ at::DimVector inversePermutation(
   }
 }
 
+template <typename T, typename CONTAINER>
+void encode_buf(T value, CONTAINER& buffer) {
+  char* v = reinterpret_cast<char*>(&value);
+  for (int i = 0; i < sizeof(T); i++) {
+    buffer.push_back(*(v++));
```

Comment:
since we care about perf, it would be preferable to avoid push_back (and resize + indexing) or at least reserve. It's also a reason not to make this over generic and stick to a container type.

Reply:
Ha, this is the part I was playing with earlier.
We do reuse the container object between iteration. So it grows as needed, but push_back shouldn't change the capacity and there shouldn't be any difference between this and (resize + indexing).

Reply:
I think this also applies to other sequential container in std... But I'm not that confident.

I agree that having a CONTAINER template argument here is ambiguous. I'll update it.